src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -509,7 +509,10 @@ private static IEnumerable<VersionRange> GetVersionRangeList(IVsReferenceItem it
         {
             char[] splitChars = new[] { ';' };
             string versionString = GetPropertyValueOrNull(item, "Version");
-
+            if (string.IsNullOrEmpty(versionString))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Error_PackageDownload_OnlyExactVersionsAreAllowed, "", item.Name));
```

Comment:
In this case and elsewhere, the error message for package ID NuGet.Common will be:

`Package 'NuGet.Common ' does not have an exact version like '[1.0.0]'. Only exact versions are allowed with PackageDownload.`

Note the trailing whitespace after the package ID.  It's unpolished.

From the start of this PR it is almost like there is an aversion to creating new resource strings.  Why not have 2 resource strings?

* one that takes package ID
* one that takes package ID and version

Reply:
@nkolev92 what do you think?

Reply:
I don't mind having separate resx files.

Reply:
I have added separate resource strings..